### PR TITLE
strings should be UTF8, but also check ASCII without crashing in dev

### PR DESCRIPTION
**Pull request by bridiver**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_ads/browser/ad_notification.cc [hunk 1]
```diff
@@ -24,14 +24,31 @@ std::unique_ptr<message_center::Notification> CreateAdNotification(
...
+                         notification_info.advertiser.length(), &advertiser)) {
+    if (base::IsStringASCII(notification_info.advertiser))
+      advertiser = base::ASCIIToUTF16(notification_info.advertiser);
+  }
+
+  base::string16 text;
+  if (!base::IsStringUTF8(notification_info.text) ||
+      !base::UTF8ToUTF16(notification_info.text.c_str(),
+                         notification_info.text.length(), &text)) {
+    if (base::IsStringASCII(notification_info.text))
```
Comment (bridiver) [inline components/brave_ads/browser/ad_notification.cc]: this is probably not necessary, but doesn't hurt anything

**Comment by bbondy** [comment]
Please remember to set merged labels and adjust the associated issue milestone.

**Comment by bbondy** [comment]
I think it only landed in c61, so marking that.

**Comment by tmancey** [comment]
@bridiver Could you please confirm label is correct, thanks

**Comment by bridiver** [comment]
0.60.x https://github.com/brave/brave-core/commit/dbaa399a439265f60912cc944dfa9b3ba18bc3e9

**Pull request merged**